File: 00-Test/src/main/java/com/leecode/高频题/_88_合并两个有序数组.java
package com.leecode.高频题;

public class _88_合并两个有序数组 {
    public void merge2(int[] nums1, int m, int[] nums2, int n) {
        /**
         * 输入：nums1 = [1,2,3,0,0,0], m = 3,
         *      nums2 = [2,5,6], n = 3
         * 输出：[1,2,2,3,5,6]
         */
        int i1 = m-1;
        int i2 = n-1;
        int cur= nums1.length-1;
        while(i2>= 0){
            if (i1>=0 && nums1[i1] > nums2[i2] ){
                nums1[cur--] = nums1[i1--];
            }else {
                nums1[cur--] = nums2[i2--];
            }
        }
    }
    public void merge3(int[] nums1, int m, int[] nums2, int n) {
        int i1 = m-1;
        int i2 = n-1;
        int cur = nums1.length-1;
        while (i2>=0){
            if (i1>=0&&nums1[i1]>nums2[i2]){
                nums1[cur--] = nums1[i1--];
            }else {
                nums1[cur--] = nums2[i2--];
            }
        }

    }


        public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i1=0, i2=0;
        int [] tmp  = new int[nums1.length];
        int cur = 0;
        while (i1 < m && i2<n){
            if (nums1[i1] < nums2[i2]){
                tmp[cur++] = nums1[i1++];
            }else {
                tmp[cur++] = nums2[i2++];
            }
        }
        while(i1 < m){
            tmp[cur++] = nums1[i1++];
        }
        while (i2<n){
            tmp[cur++] = nums2[i2++];
        }
        for (int i = 0; i < tmp.length; i++) {
            nums1[i] = tmp[i];
        }
    }

}

File: 00-Test/src/main/java/com/lee/asia/_88_合并两个有序数组.js
let merge = function(nums1, m, nums2, n) {
    //error // let i = nums1.length-1;
    // let j = nums2.length-1;
    let i = m-1;
    let j = n-1
    let cur = nums1.length-1
    while(j>=0){
        if (i>=0 && nums1[i]>nums2[j]){
            nums1[cur--]  = nums1[i--]
        }else {
            nums1[cur--] = nums2[j--]
        }
    }
    return nums1
};

let merge2 = function(nums1, m, nums2, n){
    // nums1.splice(m,nums1.length-m,...nums2)
    // nums1.sort((a,b)=>a-b)
    for(let i =1 ; i!== n ;i++){
        nums1[m+ i] = nums2[i]
    }
}

merge2([1,2,3,0,0,0],3,[2,5,6],3)

// let res = merge([1,2,3,0,0,0],3,[2,5,6],3)
let res = merge([1,2,3,0,0,0],3,[2,5,6],3)
console.log(res);